include/igl/ray_mesh_intersect.cpp

```diff
@@ -12,6 +12,41 @@ extern "C"
 #include "raytri.c"
 }
 
+namespace igl {
+template <
+  typename DerivedV,
+  typename DerivedF>
+IGL_INLINE bool ray_triangle_intersect(
+  Eigen::Vector3d s_d,
```

Comment:
should these be const references rather than non-const pass-by-copy?

Reply:
Yes. I cannot remember why I did it like that. Will fix that.

I was also not sure about your usage of `auto` and if you like named variables to give some context, even if they are basically just temporaries. Like `auto d = v.cast<double>(); foo(d)` vs. `foo(v.cast<double>())`.